src/main/java/com/emc/object/s3/S3SignerV2.java

```diff
@@ -110,6 +115,10 @@ public URL generatePresignedUrl(PresignedUrlRequest request) {
         // add signature to query string
         queryParams.put(S3Constants.PARAM_SIGNATURE, signature);
 
+        if (s3Config.getSessionToken() != null) {
```

Comment:
doesn't this need to be added **before** signing?  Also, make sure to use the constant for the parameter name

Reply:
As I mentioned in the email, ECS did not take this into the string for sign, so I add it after the signing, It works well with ECS now.